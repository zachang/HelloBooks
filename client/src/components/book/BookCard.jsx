import React from 'react';
import PropTypes from 'react-proptypes';
import { Link } from 'react-router';

/**
 * @function BookCard
 * @param {object} props
 * @return {XML} JSX
 */
const BookCard = props => (

  <div className='col l4 s10 m6 cardsm'>
    <div className='card large sticky-action'>
      <div className='card-image waves-effect waves-block waves-light'>
        <img className='activator' src={(props.book.bookImage !== null)
          ? props.book.bookImage : '../../imgs/default.jpg'}
        />
      </div>
      <div className='card-content'>
        <span className='card-title activator grey-text text-darken-4'>
          <small>{ props.book.bookName }</small>
          <i className='material-icons right'>
            more_vert
          </i>
        </span>
      </div>
      <div className='card-reveal'
        style={{
          wordWrap: 'break-word',
          backgroundColor: 'rgba(0,0,0,1)',
          color: '#FFF'
        }}
      >
        <span
          className='card-title grey-text text-darken-4'
        >
          <small>{ props.book.bookName }</small>
          <i className='material-icons right'>
            close
          </i>
        </span>
        <p>Author: { props.book.author }</p>
        <p>Category: { props.book.Category.categoryName }</p>
        <p>Published: { new Date(props.book.publishYear).getFullYear() }</p>
        <p>Pages: { props.book.pages }</p>
        <p>Stock: { props.book.bookCount }</p>
        <p>Description: { props.book.description }</p>
      </div>
      <div className='card-action home-card'>

        <Link
          to={`read/${props.book.id}`}
          className='white tooltipped hidetooltip'
          data-position='bottom'
          data-delay='50'
          data-tooltip='Read book'
        >
          <i
            className='small material-icons readBook'
            onClick={() => (props.readBook())}
            style={{ color: 'teal', cursor: 'pointer' }}
          >
            remove_red_eye
          </i>
        </Link>

        <Link
          to={`books/${props.book.id}`}
          className='white tooltipped hidetooltip'
          data-position='bottom'
          data-delay='50'
          data-tooltip='Edit book'
        >
          <i
            className='small material-icons editBook'
            style={{ color: 'teal' }}
            onClick={() => (props.editBook())}
          >
            edit
          </i>
        </Link>

        <Link
          className='white tooltipped hidetooltip'
          data-position='bottom'
          data-delay='50'
          data-tooltip='Delete book'
        >
          <i
            className='small material-icons deleteBook'
            onClick={() => (props.deleteBook(props.book.id))}
            style={{ color: 'red', cursor: 'pointer' }}
          >
            delete
          </i>
        </Link>
      </div>

    </div>
  </div>
);

BookCard.propTypes = {
  book: PropTypes.object.isRequired,
  editBook: PropTypes.func,
  deleteBook: PropTypes.func,
  readBook: PropTypes.func

};
export default BookCard;
