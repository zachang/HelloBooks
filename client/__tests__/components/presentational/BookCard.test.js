import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';

import BookCard from '../../../src/components/book/BookCard.jsx';

describe('<BookCard/>', () => {
  let wrapper;

  const props = {
    book: {
      bookImage: 'ggv.jpg',
      id: 1,
      bookName: 'Harry Potter',
      publishYear: new Date(),
      bookCount: 5,
      description: 'nice read',
      pages: 700,
      Category: {
        categoryName: 'Arts'
      }
    },
    editBook: jest.fn(),
    deleteBook: jest.fn(),
    readBook: jest.fn()
  };

  const wrapperMount = mount(<BookCard {...props} />);
  const shallowWrapper = shallow(<BookCard {...props}/>);
  const tree = toJson(shallowWrapper);


  it('renders <BookCard/> component', () => {
    expect(tree).toMatchSnapshot();
  });

  it('calls onClick', () => {
    wrapper = shallow(<BookCard {...props} />);
    wrapper.find('.readBook').simulate('click');
    expect(wrapperMount.find('.readBook').at(0).text()).toBe('remove_red_eye');
  });

  it('calls onClick', () => {
    wrapper = shallow(<BookCard {...props} />);
    wrapper.find('.editBook').simulate('click');
    expect(wrapperMount.find('.editBook').at(0).text()).toBe('edit');
  });

  it('calls onClick', () => {
    wrapper = shallow(<BookCard {...props} />);
    wrapper.find('.deleteBook').simulate('click');
    expect(wrapperMount.find('.deleteBook').at(0).text()).toBe('delete');
  });
});

