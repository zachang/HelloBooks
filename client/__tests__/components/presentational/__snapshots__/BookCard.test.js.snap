// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookCard/> renders <BookCard/> component 1`] = `
<div
  className="col l4 s10 m6 cardsm"
>
  <div
    className="card large sticky-action"
  >
    <div
      className="card-image waves-effect waves-block waves-light"
    >
      <img
        className="activator"
        src="ggv.jpg"
      />
    </div>
    <div
      className="card-content"
    >
      <span
        className="card-title activator grey-text text-darken-4"
      >
        <small>
          Harry Potter
        </small>
        <i
          className="material-icons right"
        >
          more_vert
        </i>
      </span>
    </div>
    <div
      className="card-reveal"
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,1)",
          "color": "#FFF",
          "wordWrap": "break-word",
        }
      }
    >
      <span
        className="card-title grey-text text-darken-4"
      >
        <small>
          Harry Potter
        </small>
        <i
          className="material-icons right"
        >
          close
        </i>
      </span>
      <p>
        Author: 
      </p>
      <p>
        Category: 
        Arts
      </p>
      <p>
        Published: 
        2018
      </p>
      <p>
        Pages: 
        700
      </p>
      <p>
        Stock: 
        5
      </p>
      <p>
        Description: 
        nice read
      </p>
    </div>
    <div
      className="card-action home-card"
    >
      <Link
        className="white tooltipped hidetooltip"
        data-delay="50"
        data-position="bottom"
        data-tooltip="Read book"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="read/1"
      >
        <i
          className="small material-icons readBook"
          onClick={[Function]}
          style={
            Object {
              "color": "teal",
              "cursor": "pointer",
            }
          }
        >
          remove_red_eye
        </i>
      </Link>
      <Link
        className="white tooltipped hidetooltip"
        data-delay="50"
        data-position="bottom"
        data-tooltip="Edit book"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="books/1"
      >
        <i
          className="small material-icons editBook"
          onClick={[Function]}
          style={
            Object {
              "color": "teal",
            }
          }
        >
          edit
        </i>
      </Link>
      <Link
        className="white tooltipped hidetooltip"
        data-delay="50"
        data-position="bottom"
        data-tooltip="Delete book"
        onlyActiveOnIndex={false}
        style={Object {}}
      >
        <i
          className="small material-icons deleteBook"
          onClick={[Function]}
          style={
            Object {
              "color": "red",
              "cursor": "pointer",
            }
          }
        >
          delete
        </i>
      </Link>
    </div>
  </div>
</div>
`;
